.github/workflows/test.yml

```diff
@@ -102,3 +102,44 @@ jobs:
         run: |
           ./contrib/scripts/test_localnet_liveness.sh 100 5 50 localhost
         if: env.GIT_DIFF
+
+  upgrade-test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v3.1.0
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
+      - uses: technote-space/get-diff-action@v6.0.1
+        with:
+          PATTERNS: |
+            **/**.go
+            go.mod
+            go.sum
+      - name: Install GaiaV7
```

Comment:
we can make this more  general, we can configure the upgradeFrom version and upgradeTo version 
1. by providing a label of versions, for example, upgradeFrom Version: v7.0.0. 
or 
3. maybe use present release tag -1, so if we tag this release v8.0.0, it will trigger and upgrade test from v7.0.0 to v8.0.0

here is an example of [set tag of release version as env.](https://github.com/cosmos/gaia/blob/719b81d1837c1fa189e44bea6ab12b29ed90d8ea/.github/workflows/release.yml#L16)

Reply:
let's keep that as part of a future update, i think it's still good to contain upgrades between various hard-coded versions to ensure backported changes don't break upgrades